site-src/geps/gep-724.md

```diff
@@ -0,0 +1,614 @@
+```
+
+#### Added
+```go
+type ListenerRoutes struct {
+  // ...
+  // Kinds specifies the groups and kinds of Routes that are allowed to bind to
+  // this Gateway listener. When unspecified or empty, the only limitation on
+  // the kinds of Routes supported is the Listener protocol. Kind MUST
+  // correspond to kinds of Routes that are compatible with the application
+  // protocol specified in the Listener's Protocol field. If an implementation
+  // does not support or recognize this resource type, it SHOULD set the
+  // "ResolvedRefs" condition to false for this listener with the
+  // "InvalidRoutesRef" reason.
+  //
+  // Support: Core
+  //
+  // +optional
+  Kinds []RouteGroupKind `json:"kinds,omitempty"`
+}
+
+type RouteGroupKind struct {
+  // Group is the group of the Route.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:default=networking.gateway.k8s.io
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Group *string `json:"group,omitempty"`
+	// Kind is the kind of the Route.
+	//
+	// Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Kind string `json:"kind"`
+}
+```
+
+### Routes
+
+On Routes, we remove the `RouteGateways` struct and replace it with a list of
+parent references to attach to.
+
+#### Removed
+From Route Specs:
+```go
+  // Gateways defines which Gateways can use this Route.
+  //
+  // +optional
+  // +kubebuilder:default={allow: "SameNamespace"}
+  Gateways *RouteGateways `json:"gateways,omitempty"`
+```
+
+And the structs that references:
+```go
+// RouteGateways defines which Gateways will be able to use a route. If this
+// field results in preventing the selection of a Route by a Gateway, an
+// "Admitted" condition with a status of false must be set for the Gateway on
+// that Route.
+type RouteGateways struct {
+  // Allow indicates which Gateways will be allowed to use this route.
+  // Possible values are:
+  // * All: Gateways in any namespace can use this route.
+  // * FromList: Only Gateways specified in GatewayRefs may use this route.
+  // * SameNamespace: Only Gateways in the same namespace may use this route.
+  //
+  // +optional
+  // +kubebuilder:validation:Enum=All;FromList;SameNamespace
+  // +kubebuilder:default=SameNamespace
+  Allow *GatewayAllowType `json:"allow,omitempty"`
+
+  // GatewayRefs must be specified when Allow is set to "FromList". In that
+  // case, only Gateways referenced in this list will be allowed to use this
+  // route. This field is ignored for other values of "Allow".
+  //
+  // +optional
+  GatewayRefs []GatewayReference `json:"gatewayRefs,omitempty"`
+}
+```
+
+#### Added
+To Route Specs:
+```go
+  // AttachTo defines the parent resources this Route should be attached to. The
+  // only kind of target resource with "Core" support is Gateway. This API may
+  // be extended in the future to support additional kinds of parent resources
+  // such as Routes.
```

Comment:
Just want to call out (no action required): This is not the case today because RouteStatus explicitly reference Gateway.

Question: Should we rename fields in the status block as well?

Reply:
Good call, added a section renaming this.